src/runtime/contrib/nvshmem/init.cc

```diff
@@ -106,12 +106,26 @@ void InitNVSHMEMWrapper(String args) {
   InitNVSHMEM(uid_64, num_workers, worker_id_start);
 }
 
+void NVSHMEMXCumoduleInit(void* cuModule) {
+  CUmodule mod = static_cast<CUmodule>(cuModule);
+  auto status = nvshmemx_init_status();
+  // The NVSHMEM library must have completed device initialization prior to
+  // nvshmemx_cumodule_init. If not, we skip the cumodule initialization.
```

Comment:
if not device initialized, we should return with error

Reply:
The design here is to enable NVSHMEM compilation and linking broadly for every kernel, including those whose NVSHMEM context is not initialized and do not use NVSHMEM in their kernels.

In such case, `nvshmemx_init_status()` is used to check whether we need to call `nvshmemx_cumodule_init` or not. If not device initialized, we just skip `nvshmemx_cumodule_init`.